tasks/eva_2835/gather_per_species_per_assembly_variant_remapping_counts.py

```diff
@@ -0,0 +1,311 @@
+        tax_assembly[tax_id] = assembly_accession
+
+    # Collect statistics for each taxonomy and each assembly
+    for taxid, assembly in tax_assembly.items():
+        for val in range(len(assembly)):
+            columns, values, taxid, assembly_accession = gather_counts_per_tax_per_assembly(remapping_root_path,
+                                                                                            taxid, assembly[val])
+
+            # Adding the fields and values to the master list for each taxonomy and assembly
+            all_columns_for_specific_taxid_assembly.append(columns)
+            all_values_for_specific_taxid_assembly.append(values)
+
+            # Adding the taxonomy and assembly accession for a particular file to the master list
+            all_tax_id.append(taxid)
+            all_assembly.append(assembly_accession)
+
+            # Updating the final output column in a sorted order every time from an individual yml file
+            all_columns.extend(columns)
+            all_columns = list(dict.fromkeys(all_columns))
+            all_columns.sort()
+
+    # Returning the final list of output columns along with the master lists of columns and
+    # their corresponding values for a specific tax id and assembly accession
+    return all_columns, all_columns_for_specific_taxid_assembly, all_values_for_specific_taxid_assembly, \
+           all_tax_id, all_assembly
+
+
+def gather_counts_per_tax_per_assembly(path, taxid, assembly_accession):
+    """
+    This function is used to store the counts of the remapped variants along wih the reason for failures
+    at different rounds (i.e. for different lengths of the flanking region) for a particular taxonomy and
+    assembly
+
+    Input: The taxonomy id and the assembly accession from generate_output() along with the input files full
+           path
+
+    Output: All the statistics i.e. columns and values for a particular taxonomy and assembly
+
+    """
+
+    # Setting the filename for the eva counts
+    filename_eva = str(assembly_accession) + "_eva_remapped_counts.yml"
+    filename_eva = os.path.join(path, str(taxid), str(assembly_accession), "eva", filename_eva)
+
+    # Setting the filename for the dbsnp counts
+    filename_dbsnp = str(assembly_accession) + "_dbsnp_remapped_counts.yml"
+    filename_dbsnp = os.path.join(path, str(taxid), str(assembly_accession), "dbsnp", filename_dbsnp)
+
+    # Calling functions for gathering stats per file from eva and dbsnp data and storing the data in a dictionary
+    # as a key-value pair
+    columns_values = gather_counts_per_file(filename_eva)
+    columns_values_dbsnp = gather_counts_per_file(filename_dbsnp)
+
+    # Adding the values of dbsnp and eva with the common columns i.e. fields
+    for column in columns_values_dbsnp:
+        if column in columns_values:
+            columns_values_dbsnp[column] = columns_values_dbsnp[column] + columns_values[column]
```

Comment:
I find it a bit counter-intuitive to accumulate the values in `column_values_dbsnp` rather than the more generically named `column_values` (or into a third dict altogether)...

Reply:
I have made this change in the revised code